python/cugraph/structure/graph.py

```diff
@@ -103,6 +104,51 @@ def clear(self):
         self.adjlist = None
         self.transposedadjlist = None
 
+    def add_nodes_from(self, nodes, bipartite='nodes'):
+        """
+        Add nodes information to the Graph.
+        If bipartite argument is passed by the user, the graph is considered
+        bipartite and the entered nodes are saved as the set of nodes in one
+        partition.
+        If bipartite argument is not passed, the nodes are saved as information
+        and Graph is not considered to be bipartite.
+        """
+        set_names = [i for i in self._nodes.keys() if i != 'nodes']
+        if len(set_names) < 2:
+            if bipartite != 'nodes':
+                self.bipartite = True
+            self._nodes[bipartite] = cudf.Series(nodes)
```

Comment:
Should we allow for appending additional nodes to one of the bipartite sets?  This code would do the following, if I interpret it correctly:
 * If you call it with bipartite = 'A' twice, the second call would overwrite the first call
 * If you call it with bipartite = 'A', then call it with bipartite = 'B', then call it with bipartite = 'A' it would fail because it would think we're adding a third type.

Also, with Brad's comment... might want to be able to support multi-partite, so perhaps checking the length of set_names is unnecessary.  is_bipartite would be true if there are exactly 2 keys in the _nodes map.

Reply:
Good catch. Updated to fix this discrepancy. Now the code always follows "If you call it with bipartite = 'A' twice, the second call would overwrite the first call"